src/Utilities/ToolTask.cs

```diff
@@ -1715,6 +1715,38 @@ public override bool Execute()
         } // Execute()
 
         /// <summary>
+        /// Replace backslashes with OS-specific path separators,
+        /// except when likely that the backslash is intentional.
+        /// </summary>
+        /// <remarks>
+        /// Not a static method so that an implementation can
+        /// override with more-specific knowledge of what backslashes
+        /// are likely to be correct.
+        /// </remarks>
+        virtual protected string ReplaceBackslashes(string input)
+        {
+            StringBuilder sb = new StringBuilder(input);
+
+            int length = sb.Length;
+
+            for (int i = 0; i < length; i++)
+            {
+                // Backslashes must be swapped, because we don't
+                // know what inputs are paths or path fragments.
+                // But it's a common pattern to have backslash-escaped
+                // quotes inside quotes--especially for the VB /define:
+                // parameter. So don't replace a backslash immediately
+                // followed by a quote.
+                if (sb[i] == '\\' && (i == length - 1 || sb[i + 1] != '"'))
```

Comment:
What are all the escapable characters? Like space, double quotes, single quotes, backticks, piping characters?. Is it worth doing the fix for them as well?